### PR TITLE
[DT-1012] Heap tracking for reverse resolution endpoints

### PR DESCRIPTION
**Pull request by tunguyennnnn**
## Background

[DT-1012] https://linear.app/unstoppable-domains/issue/DT-1012/add-heap-tracking-for-resolution-services-reverse-api-endpoints

## Changes

* Add heap tracking for `/reverse/:address` and `reverse/query` endpoints

## Code Review

This Pull Request was thoroughly reviewed and meets all Code Review Standards pertaining to:

- [ ] **Implementation** - satisfied acceptance criteria
- [ ] **Communication** - notified engineers/stakeholders about impactful changes
- [ ] **Error handling** - handled, logged & alerted on errors
- [ ] **Documentation** - wrote readable code & commits, documented unintuitive code
- [ ] **Testing** - unit & E2E test coverage, tested in staging, DB migrations backwards compatible
- [ ] **Security** - sanitized inputs, vetted dependencies, validated/secured API requests, no committed secrets
- [ ] **Performance** - optimized expensive algorithms & database queries

Please select all **applied** standards relevant to this PR.
## Confirmation

- [ ] **Assignee Confirmation** - I (the author) have filled out the template above
- [ ] **Reviewer Confirmation** - I (a/the reviewer) confirm 1) the author has filled out the template and checked the box that says **Assignee Confirmation** 2) I reviewed the code and selected all applicable items in the code review section.

### CODE REVIEW

File: src/types/heap.ts [hunk 1]
```diff
@@ -5,6 +5,8 @@ export enum HeapEvents {
   GET_DOMAINS = 'rsbe - get - domains list',
   GET_LATEST_DOMAIN_TRANSFER = 'rsbe - get - domain transfer',
   GET_DOMAIN_RECORDS = 'rsbe - get - domain records',
+  GET_REVERSE = 'rsbe - get - reverse resolution',
+  POST_BULK_REVERSE = 'rbse - get - reverse resolution list',
```
Comment (DChan0319) [inline src/types/heap.ts:9]: rsbe

**Review by DChan0319 — CHANGES_REQUESTED** [review]

File: src/controllers/ReverseController.test.ts [hunk 2]
```diff
@@ -282,6 +304,17 @@ describe('ReverseController', () => {
           },
         ],
       });
+
+      expect(trackStub).to.be.calledWith({
+        identity: SUPERTEST_TESTING_IP,
+        eventName: HeapEvents.POST_BULK_REVERSE,
+        properties: {
```
[the remark this answers is not in the document]
Comment (tunguyennnnn) [reply src/controllers/ReverseController.test.ts:311]: we only process addresses right? if they pass junk data to us we should just ignore.

File: src/types/heap.ts (hunk 1 above)
Comment (tunguyennnnn) [reply src/types/heap.ts]: good catch 🙏!

File: src/controllers/ReverseController.test.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (tunguyennnnn) [reply src/controllers/ReverseController.test.ts:312]: ah yea I see what you mean, yea I added `response_domain_name` and `response_domain_names` to the heap events

**Review by DChan0319 — APPROVED** [review]
looks good!

**Pull request merged**